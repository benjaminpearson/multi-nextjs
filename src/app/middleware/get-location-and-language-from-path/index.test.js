import getLocationAndLanguageFromPath from '.';

jest.mock('../../cookie');

describe('server', () => {
  it('gets location and language from valid path and returns adjusted asPath', () => {
    const req = { url: '/au/en/feature' };
    expect(getLocationAndLanguageFromPath({ req, res: {} })).toMatchSnapshot();
    // @TODO: Check cookie set
  });

  it('returns false if invalid location and language', () => {
    const req = { url: '/feature' };
    expect(getLocationAndLanguageFromPath({ req, res: {} })).toBe(false);
    // @TODO: Check cookie not set
  });
});

describe('client', () => {
  it('gets location and language from valid path and returns adjusted asPath', () => {
    expect(getLocationAndLanguageFromPath({ asPath: '/au/en/feature' })).toMatchSnapshot();
    // @TODO: Check cookie set
  });

  it('returns false if invalid location and language', () => {
    expect(getLocationAndLanguageFromPath({ asPath: '/feature' })).toBe(false);
    // @TODO: Check cookie not set
  });
});
