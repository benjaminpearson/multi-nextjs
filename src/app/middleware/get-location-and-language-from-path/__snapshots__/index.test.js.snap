// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client gets location and language from valid path and returns adjusted asPath 1`] = `
Object {
  "asPath": "/feature",
  "language": "en",
  "location": "au",
}
`;

exports[`server gets location and language from valid path and returns adjusted asPath 1`] = `
Object {
  "asPath": "/feature",
  "language": "en",
  "location": "au",
}
`;
